File: src/main/scala/org/nypl/mss/GumshoePackage.scala
package org.nypl.mss

import java.io.{FileInputStream, File}
import org.apache.commons.io.FilenameUtils
import scala.collection.mutable
import scala.io.Source

class GumshoePackage(packageLoc: String) {
  val files: File = new File(packageLoc)
  var indexFile: File = null
  var parserFiles: File = null
  val configMap = new mutable.HashMap[String, String]()

  for(file <- files.listFiles()){
    if(FilenameUtils.getExtension(file.getName) == "tsv"){
      indexFile = file
    } else if(FilenameUtils.getExtension(file.getName) == "txt"){
      for(line <- Source.fromInputStream(new FileInputStream(file)).getLines()){
        val kv = line.split(":\\s*")
        configMap += (kv(0) -> kv(1))
      }
    } else if(file.getName == "files" && file.isDirectory && file.listFiles.size > 1){
      parserFiles = file
    }
  }

  private def checkValid() : Boolean = {
    if(indexFile != null && parserFiles != null) true else false
  }

  override def toString() : String = {
    val sb = new mutable.StringBuilder()

    sb.append(files.getName() + "\nfiles: " + parserFiles.getAbsolutePath + "\nindex: " + indexFile.getAbsoluteFile
    + "\nconfiguration:")
    for(kv <- configMap){
      sb.append("\n\t" + kv._1 +": " + kv._2);
    }
    sb.toString
  }

}

File: src/main/java/org/nypl/mss/gumshoesolrizerj/GumshoePackage.java
package org.nypl.mss.gumshoesolrizerj;


import java.io.BufferedReader;
import java.io.FileReader;
import java.io.File;
import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

import org.apache.commons.io.FilenameUtils;

public class GumshoePackage {
    private File files, indexFile, parserFiles;
    private Map<String, String> configMap = new HashMap();

    GumshoePackage(String packageLoc) throws IOException {
        files = new File(packageLoc);
        for(File file: files.listFiles()){
            if(FilenameUtils.getExtension(file.getName()).equals("tsv")){
                indexFile = file;
            } else if(FilenameUtils.getExtension(file.getName()).equals("txt")){
                BufferedReader br = new BufferedReader(new FileReader(file));
                String line;
                while((line = br.readLine()) !=null){
                    String[] kv = line.split(":\\s*");
                    configMap.put(kv[0].trim(), kv[1].trim());
                }
            } else if(file.getName().equals("files") && file.isDirectory()&& file.listFiles().length >= 1){
                parserFiles = file;
            }
        }
    }

    public boolean checkValid(){
        if(indexFile != null && parserFiles != null)
            return true;
        else return false;
    }

    @Override
    public String toString(){
        StringBuilder sb = new StringBuilder();

        sb.append(files.getName()
                + "\nfiles: " + parserFiles.getAbsolutePath()
                + "\nindex: " + indexFile.getAbsoluteFile()
                + "\nconfiguration:");

        for(Map.Entry e: configMap.entrySet()){
            sb.append("\n\t" + e.getKey().toString() +": " + e.getValue().toString());
        }
        return sb.toString();
    }

    public File getIndexFile(){
        return indexFile;
    }

    public File getParserFiles(){
        return parserFiles;
    }

    public HashMap<String, String> getConfigMap(){
        return (HashMap<String, String>) configMap;
    }
}
